audformat/core/utils.py

```diff
@@ -196,6 +196,50 @@ def concat(
         return df
 
 
+def duration(
+        obj: typing.Union[pd.Index, pd.Series, pd.DataFrame],
+) -> float:
+    r"""Duration of all segments present in a segmented object.
+
+    Args:
+        obj: object conform to
+            :ref:`table specifications <data-tables:Tables>`
+            and containing a segmented index
+
+    Returns:
+        duration in seconds
+
+    Raises:
+        ValueError: if ``obj`` does not contain a segmented index
+
+    Example:
+
+        >>> idx = segmented_index(
+        ...     files=['a', 'b', 'c'],
+        ...     starts=[0, 1, 3],
+        ...     ends=[1, 2, 4],
+        ... )
+        >>> duration(idx)
+        3.0
+
+    """
+    if index_type(obj) != define.IndexType.SEGMENTED:
```

Comment:
We could add support for filewise tables if we use `audiofile.duration()`.

Reply:
I was thinking about that, but I see two problems:

1. it will only work if the files are available
2. it can take a very long time

For that reason I thought to restrtict `duration` here only to segmented tables and use a function inside `audb` to get duration for filewise tables as the duration is stored in the dependency tables generated with `audiofile.duration`.
Of course it would be nicer to have just one function inside `audformat`, but I don't see an easy solution for it.

Reply:
Yes, in `audb` we should of course use the information from the dependency table, but outside of `audb` it could still be useful to support `filewise` tables, too. That it may take a long time is not really an argument against it I think. But is in fact problematic is that files have to available. I think at the moment there is only one other function - `utils.to_filewise()` - that also assumes that files exists. All other functions do not start from this assumption. So I think it's a nice to have, but we can also skip it for now.

Reply:
Btw: if we don't support `filewise` tables, the user may try as a workaround:

```python
df = db['table'].df
df = audformat.utils.to_segmented_index(df)
dur = audformat.utils.duration(df)
```

But currently that returns `0.0`. So I think we anyway have to use `audiofile.duration()` where `end=None`.

Reply:
Or should we move this functino altogether to `audb` as suggested in https://gitlab.audeering.com/tools/audb2/-/issues/18#note_99752 ?

Reply:
Yes, I think I would vote for that. I find it a bit limiting that we return `NaN` if a segment end is `NaT` and that we don't have support `filewise` tables. On the other hand, we can not safely use `audiofile.duration()` if the files do not exist.

Reply:
Agreed. I have marked this merge request as WIP, and will close it after we have added the functionality to `audb`.

Reply:
I support now filewise and segmented index (also with `NaT` here).
`audb.info.duration()` can still be used for a complete database, and `audformat.utils.duration()` might be handy if we want to expand `audb.info.duration()` for specific tables, see https://github.com/audeering/audb/issues/106